### PR TITLE
Trying to do ascii file injections in pyCBC 

### PR DESCRIPTION
**Pull request by shubha89**
For enabling ascii file injections I did the following thing

 - I made changes into waveform.py to add function called get_waveform_from_ascii, this function takes the ascii file name and give out hp and hc in numpy array

-  then I went ahead and did the changes to the inject.py defining a class _AXMLInjectionSet and this is activated if we give the injection xml file as xmlgz (for .xml or .xml.gz it uses the class _XMLInjectionSet ) I changed that in InjectionSet class. I know this is a very hacky way to do this but I needed the features of inspinj to get some nice distance distributions.

I have a few questions/requests 

- Do I need to change more things that I not aware of to use this function properly ?
- Do you have some suggestions to make this more useable ?
- If we have time can we please check these changes and see if the things i did makes sense ?

### CODE REVIEW

File: pycbc/waveform/waveform.py [hunk 1]
```diff
@@ -927,6 +927,37 @@ def get_waveform_filter_length_in_time(approximant, template=None, **kwargs):
     else:
         return None
 
+def get_waveform_from_ascii(waveform, deltaT, epoch=0.0):
```
Comment (ahnitz) [inline pycbc/waveform/waveform.py]: The "waveform" argument doesn't seem to be used. Why is it needed? Also, is there no way to get delta t from the txt file? Is there a way to get this information from whatever is in the first column?

Comment (shubha89) [reply pycbc/waveform/waveform.py]: The waveform argument supposedly contains the path to the text file (but lalapps_inspinj doesnt allow me to do that) 
Yes we can simply calculate the delta T from the time column its simple

File: pycbc/inject/inject.py [hunk 2]
```diff
@@ -455,11 +457,14 @@ class InjectionSet(object):
...
     def __init__(self, sim_file, **kwds):
         ext = os.path.basename(sim_file)
-        if ext.endswith(('.xml', '.xml.gz', '.xmlgz')):
-            self._injhandler = _XMLInjectionSet(sim_file, **kwds)
+        if ext.endswith(('.xml', '.xml.gz')):
+            self._injhandler = _XMLInjectionSet(sim_file, **kwds) 
             self.indoc = self._injhandler.indoc
         elif ext.endswith(('hdf', '.h5')):
             self._injhandler = _HDFInjectionSet(sim_file, **kwds)
+        elif ext.endswith(('.xmlgz')): ## If the injection is based on ASCII file and uses the distribution of the inspinj, please use this extention (sorry for this terrible way)
```
Comment (ahnitz) [inline pycbc/inject/inject.py]: Why not just add the switch based on the waveform into the xml injection set? Is there a reason we need to duplicate the class?

Comment (shubha89) [reply pycbc/inject/inject.py]: Agreed, added in the same class a condition, if the approximant name is 'ascii' it takes injections from ascii file

File: pycbc/inject/inject.py [hunk 3]
```diff
@@ -403,15 +404,22 @@ def make_strain_from_inj_object(self, inj, delta_t, detector_name,
...
-        name, phase_order = legacy_approximant_name(inj.waveform)
-
-        # compute the waveform time series
-        hp, hc = get_td_waveform(
-            inj, approximant=name, delta_t=delta_t,
-            phase_order=phase_order,
-            f_lower=f_l, distance=inj.distance,
-            **self.extra_args)
+	approx_name=inj.waveform #put the path to the ASCII file as approximant
+	if approx_name.endwith(('.txt', '.dat')):
```
Comment (ahnitz) [inline pycbc/inject/inject.py]: I don't think this method exists. That is why travis is failing the unittests.

File: pycbc/waveform/waveform.py [hunk 4]
```diff
@@ -927,6 +927,36 @@ def get_waveform_filter_length_in_time(approximant, template=None, **kwargs):
     else:
         return None
 
+def get_waveform_from_ascii(waveform,epoch=0.0,Norm=5.9029581035870565e+20):
```
Comment (ahnitz) [inline pycbc/waveform/waveform.py]: Can you make sure to follow PEP8? That means spaces after commas, and use only lower case variable names.

File: pycbc/waveform/waveform.py [hunk 5]
```diff
@@ -927,6 +927,36 @@ def get_waveform_filter_length_in_time(approximant, template=None, **kwargs):
...
+    data=numpy.loadtxt(infile)
+    t=data[:,0]
+    hp1=data[:,1]
+    hc1=data[:,2]
+    deltaT=(t[3]-t[2])
+    hp_ref=hp1/N
+    hc_ref=hc1/N
+    hp=TimeSeries(hp_ref, delta_t=deltaT, epoch=epoch)
+    hc=TimeSeries(hc_ref, delta_t=deltaT, epoch=epoch)
+    return hp, hc, 1./deltaT
```
Comment (ahnitz) [inline pycbc/waveform/waveform.py]: I don't think you need to return 1.0 / deltaT here. That information is already contained in the waveform themselves. I.E. as hp.duration

File: pycbc/waveform/waveform.py [hunk 6]
```diff
@@ -927,6 +927,36 @@ def get_waveform_filter_length_in_time(approximant, template=None, **kwargs):
...
+    sample rate : retunrs the sample rate of the wavefom in the txt file 
+    """
+    infile=waveform
+    #'/home/shubhanshu.tiwari/cWB_Sergey_Shubhanshu_Vaibhav/cWB_share_m1-1.0000000e+01_m2-1.0000000e+01_ecc-3.0000000e-01.dat'
+    N=Norm # 5.9029581035870565e+20 ## at 100 Mpc change for other waveforms
+    data=numpy.loadtxt(infile)
+    t=data[:,0]
+    hp1=data[:,1]
+    hc1=data[:,2]
+    deltaT=(t[3]-t[2])
```
Comment (ahnitz) [inline pycbc/waveform/waveform.py]: again, make sure to use lower case variable names

File: pycbc/inject/inject.py [hunk 7]
```diff
@@ -403,15 +404,22 @@ def make_strain_from_inj_object(self, inj, delta_t, detector_name,
...
+		hp, hc, delta_t_ascii = get_waveform_from_ascii(inj.waveform)
+		if delta_t != delta_t_ascii:
+			raise ValueError("Sample rate of the ascii and the ini are different")
+		else:
+			hp = (hp/inj.distance)*100000 #reference distance for ASCII waveform is 100Mpc
```
Comment (ahnitz) [inline pycbc/inject/inject.py]: Any reason we can't set the distance as normal? I.E. normally distance is taken to be in Kpc for the xml format, and internally we use Mpc for everything. Why is this simply not * 100 ? Any reason the ASCII format needs to have this distance convention?

File: pycbc/inject/inject.py (hunk 3 above)
Comment (spxiwh) [reply pycbc/inject/inject.py]: `endwith` -> `endswith` I think

**Comment by ahnitz** [comment]
@shubha89 Can you rebase this so we can make sure the tests pass, otherwise, I think we can merger this.

**Comment by spxiwh** [comment]
Closing this due to long inactivity. The pull request can be easily re-opened if this is still something that we want.

**Pull request closed without merging**